kinvolk/headlamp
# Add clusters dynamically

- [X] Backend allows to set up a cluster dynamically
- [X] Clusters set up dynamically show up in consequent calls to /config
- [x] Backend has tests for the new functionality
- [x] Plugins can add a cluster dynamically (we have a way to dispatch such a request from the frontend)

fixes #456 

## How to test

- Set up a new cluster, and remove details from your `~/.kube/config` 
- add the cluster details to the `plugins/examples/dynamic-clusters/src/index.tsx` source
- cd `plugins/examples/dynamic-clusters/ && npm start` the plugin
- start headlamp
- click the "New Cluster" button in the app bar at the top
- headlamp should reload with the new cluster
